### PR TITLE
Update ccall syntax to `@ccall`

### PR DESCRIPTION
**Pull request by lgoettgens**
this PR was created by applying the following script to all relevant files.
I'll leave out some files to avoid conflicts with currently open PRs. To remember them for the future, the list of skipped files is:
- ~~`src/calcium/ca.jl`, `src/calcium/qqbar.jl` due to https://github.com/Nemocas/Nemo.jl/pull/1893~~
- `src/flint/fq_abs_series.jl`, `src/flint/fq_rel_series.jl` due to https://github.com/Nemocas/Nemo.jl/pull/1929

```julia
const pattern = r"""\(?ccall\s*\(\s*\(\s*:([a-zA-Z0-9_]+)\s*,\s*([a-zA-Z0-9_]+)\s*\),\s*([a-zA-Z0-9{}$()_]+)\s*,\s*\((([^()]|\{\(|\)\})*)\)\s*,\s*(.*)\s*\)\)?"""

function replacement(str)
    if first(str) == '(' && str[end-1:end] == "))"
        str_ = str[2:end-1]
    else
        str_ = str
    end
    match = Base.match(pattern, str_)
    fun_name = match.captures[1]
    lib_name = match.captures[2]
    ret_type = match.captures[3]
    arg_types = filter(!isempty, strip.(split(match.captures[4], ",")))
    args = filter(!isempty, strip.(split(match.captures[end], ",")))
    if length(arg_types) != length(args)
        @info "Error: number of argument types and arguments do not match" fun_name lib_name ret_type arg_types args
        return str
    end
    for arg in args
        if count('(', arg) != count(')', arg)
            @info "Error: unbalanced parentheses in argument" fun_name lib_name ret_type arg_types args
            return str
        end
    end

    # Constructing new syntax for arguments
    arg_str = join(["$(maybe_parentheses(arg))::$(arg_type)" for (arg, arg_type) in zip(args, arg_types)], ", ")
    
    ret = "@ccall $(lib_name).$(fun_name)($(arg_str))::$(ret_type)"
    if str != str_
        ret = "($ret)"
    end
    return ret
end

function maybe_parentheses(arg)
    if first(arg) == '(' && last(arg) == ')'
        return arg
    end
    if occursin("+", arg) || occursin("-", arg)
        return "($arg)"
    else
        return arg
    end
end

function transform_ccalls(file::String)
    # Read the content of the input file
    content = read(file, String)

    transformed_content = replace(content, pattern => replacement)

    open(file, "w") do f
        write(f, transformed_content)
    end
end

```

ping @fingolfin

### CODE REVIEW

**Comment by thofma** [comment]
OK, but why?

File: .git-blame-ignore-revs [hunk 1]
```diff
@@ -1,2 +1,5 @@
 # Adjusted indentation to 2 spaces
 ba3e1355fe8b2612b1f48d26ab68f6902efde691
+
+# Replaced many `ccall` by `@ccall`
+4850d2b8e7b3c6da47836a926f727bfabe4a6654
```
Comment (fingolfin) [inline .git-blame-ignore-revs:5]: So I guess for that to work we'd need to "merge" (not rebase, not squash)

**Review by fingolfin — APPROVED** [review]
Fine by me

**Comment by lgoettgens** [comment]
Since https://github.com/Nemocas/Nemo.jl/pull/1893 is basically finished, I'll update this PR later today to include the two left out files as well

File: .git-blame-ignore-revs (hunk 1 above)
Comment (lgoettgens) [reply .git-blame-ignore-revs:5]: I dropped this for now. It seems easier to do another small PR afterwards than to keep this hash right during all of the rebasing I am during currently.

**Pull request merged**